### PR TITLE
Added .NetCore 3.0 to IntegrationAspNetCore and updated derived proje…

### LINKED ISSUE #3213: Json Compat issues with Skills and .Net Core 3.1
We have identified some issues when root bots (skill consumers) try to consume bots written in .Net Core 2.x or 3.x.

Here are the two issues we've found so far:

1. A skill consumer throws a MissingMethodException when receiving a response from a skill
image

2. OAuth Cards render empty in a 3.1 Core skill consumer.

We are currently researching the issue and will post an update once we find a resultion.

Related articles:

[Breaking changes to Microsoft.AspNetCore.App in 3.0](https://github.com/aspnet/Announcements/issues/325)
[Use the ASP.NET Core shared framework](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-3.0&tabs=visual-studio#use-the-aspnet-core-shared-framework)
[Discussion: JsonResult has moved to the Microsoft.AspNetCore.Mvc.Core assembly](https://github.com/aspnet/AspNetCore/issues/14176)
[The future of JSON in .NET Core 3.0](https://github.com/dotnet/announcements/issues/90)

FYI: @johnataylor & @lzc850612 & @lauren-mills

### PR DESCRIPTION
**Pull request by gabog**
Added .NetCore 3.0 to IntegrationAspNetCore and updated derived projects.
Fixes #3213

### CODE REVIEW

File: libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Microsoft.Bot.Builder.Integration.AspNet.Core.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.0</TargetFrameworks>
```
Comment (tomlm) [inline libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Microsoft.Bot.Builder.Integration.AspNet.Core.csproj]: >netstandard2.0;netcoreapp3.0< [](start = 22, length = 29)

I think this should be <TargetFrameworks>netcoreapp2.1;netcoreapp3.0</TargetFrameworks>

File: libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Microsoft.Bot.Builder.Integration.AspNet.Core.csproj [hunk 2]
```diff
@@ -23,17 +23,23 @@
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="AsyncUsageAnalyzers" Version="1.0.0-alpha003" PrivateAssets="all" />
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'netcoreapp2.1'">
```
Comment (tomlm) [inline libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Microsoft.Bot.Builder.Integration.AspNet.Core.csproj]: >netstandard2.0' [](start = 49, length = 15)

remove this clause?

**Comment by tomlm** [comment]
>     <DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Builder.Integration.AspNet.Core.xml</DocumentationFile>

?? this doesn't feel right.

---
Refers to: libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Microsoft.Bot.Builder.Integration.AspNet.Core.csproj:18 in 34bb751. [](commit_id = 34bb7510a35759349efaff9a8c7ef7382cf24aa5, deletion_comment = False)

File: tests/integration/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests.csproj [hunk 3]
```diff
@@ -7,9 +7,16 @@
     <Configurations>Debug;Release</Configurations>
   </PropertyGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+    <FrameworkReference  Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'netcoreapp2.1'">
```
Comment (tomlm) [inline tests/integration/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests.csproj]: >Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'netcoreapp2.1'" [](start = 13, length = 95)

don't think you need a condition here.

File: tests/Microsoft.Bot.Builder.TestBot.Tests/Microsoft.Bot.Builder.TestBot.Tests.csproj [hunk 4]
```diff
@@ -19,6 +18,14 @@
     </Content>
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.TestBot.Tests/Microsoft.Bot.Builder.TestBot.Tests.csproj]: >FrameworkReference [](start = 5, length = 18)

??? This seems like a mistake?

File: tests/Microsoft.Bot.Streaming.Tests/Microsoft.Bot.Streaming.Tests.csproj [hunk 5]
```diff
@@ -9,13 +9,20 @@
     <Compile Remove="ReceiveBaseTests.cs" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+    <FrameworkReference  Include="Microsoft.AspNetCore.App" />
```
Comment (tomlm) [inline tests/Microsoft.Bot.Streaming.Tests/Microsoft.Bot.Streaming.Tests.csproj]: >FrameworkReference [](start = 5, length = 18)

mistake?

**Comment by gabog** [comment]
I will address Tom's comments in a new PR where we will also include some fixes for gov cloud and skills. Closing this PR for now.

**Pull request closed without merging**